### ISSUE TITLE
Tests flaking on tool artifact extraction on Windows

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#126438]
[This issue](https://github.com/flutter/flutter/issues/123519) is happening again on CI.

https://github.com/flutter/flutter/issues/126408
https://github.com/flutter/flutter/issues/126410
https://github.com/flutter/flutter/issues/126418

cc folks from the previous instance: @gspencergoog @christopherfujino @Hixie. From the thread on the previous GitHub issue, it sounds like there may be new logging to examine. Provisionally assigning to @christopherfujino to investigate that, and marking P1 to match the elevated flake rate.

**Comment by christopherfujino** [comment]
@christopherfujino you should try to invoke a powershell script to do a stat on the file to see if we can learn anything useful.

**Comment by christopherfujino** [comment]
Here's a recent LUCI build: https://ci.chromium.org/ui/p/flutter/builders/prod/Windows_android%20basic_material_app_win__compile/10017/overview

```
[2023-05-06 13:29:02.117368] [STDOUT] stderr: Flutter could not download and/or extract https://storage.googleapis.com/flutter_infra_release/flutter/23f730efbff723f577fb9e59ca3d88f7ec011e76/android-x64-release/windows-x64.zip. Ensure you have network connectivity and all of the required dependencies listed at flutter.dev/setup.
[2023-05-06 13:29:02.117368] [STDOUT] stderr: The original exception was: FileSystemException: Cannot open file, path = 'C:\opt\s\w\ir\x\w\recipe_cleanup\tmpx0hvlp3t\flutter sdk\bin\cache\downloads\storage.googleapis.com\flutter_infra_release\flutter\23f730efbff723f577fb9e59ca3d88f7ec011e76\android-x64-release\windows-x64.zip' (OS Error: Operation did not complete successfully because the file contains a virus or potentially unwanted software.
[2023-05-06 13:29:02.117368] [STDOUT] stderr: , errno = 225).
```

**Comment by christopherfujino** [comment]
@yusuf-goog did we ever set up bot-level defender denylists?

**Comment by yusuf-goog** [comment]
@christopherfujino I checked on our windows bot, there are no threats detected and no deny-lists in place.

**Comment by christopherfujino** [comment]
> @christopherfujino I checked on our windows bot, there are no threats detected and no deny-lists in place.

how did you check for detected threats, and could we incorporate that into a device doctor check?

**Comment by yusuf-goog** [comment]
I manually looked on the machine Windows security settings, and no threats were reported.

**Comment by christopherfujino** [comment]
> @yusuf-goog could you help get the list of threat list from PowerShell commands?

friendly ping @yusuf-goog

**Comment by yusuf-goog** [comment]
Let me dig into this more. Last check, there was no exceptions being reported.

**Comment by christopherfujino** [comment]
> @christopherfujino I'm able to reproduce a similar issue roughly once a week on my machine when extracting the Dart SDK. It seems 7z hangs as it prompts on whether I'd like to override existing files.
> 
> 7z output when it hangs on Dart SDK extraction...
> I updated the `update_dart_sdk.ps1` script to print 7z output and got:
> 
> ```
> Checking Dart SDK version...
> Downloading Dart SDK from Flutter engine 785b24900a360ed872d1f779a65d6c062ce69448...
> Expanding downloaded archive...
> 
> 7-Zip 22.01 (x64) : Copyright (c) 1999-2022 Igor Pavlov : 2022-07-15
> 
> Scanning the drive for archives:
> 1 file, 217759634 bytes (208 MiB)
> 
> Extracting archive: C:\Code\f\flutter\bin\cache\dart-sdk-windows-x64.zip
> --
> Path = C:\Code\f\flutter\bin\cache\dart-sdk-windows-x64.zip
> Type = zip
> Physical Size = 217759634
> 
> 
> Would you like to replace the existing file:
>   Path:     C:\Code\f\flutter\bin\cache\LICENSE.dart_sdk_archive.md
>   Size:     579 bytes (1 KiB)
>   Modified: 2023-08-14 13:05:02
> with the file from archive:
>   Path:     LICENSE.dart_sdk_archive.md
>   Size:     579 bytes (1 KiB)
>   Modified: 2023-08-14 13:05:02
> ? (Y)es / (N)o / (A)lways / (S)kip all / A(u)to rename all / (Q)uit?
> ```
> 
> I suspect the [step to rename the previous Dart SDK](https://github.com/flutter/flutter/blob/51a0ec006fbac207ad976dd5423bbb87ec3929de/bin/internal/update_dart_sdk.ps1#L53-L56) failed, was skipped, or was run concurrently (perhaps the Flutter lock wasn't properly acquired?). One solution might be to overwrite files when extracting from the Dart SDK zip. For example, we could use the [`-aoa` switch](http://web.mit.edu/outland/arch/i386_deb50/build/p7zip_4.44/DOCS/MANUAL/switches/overwrite.htm) on 7z.

To my uneducated eye, this looks unrelated to:

```
Operation did not complete successfully because the file contains a virus or potentially unwanted software
```

Or are you suggesting either the root cause or fix may be the same?

**Comment by loic-sharma** [comment]
Ah I missed that log line in your reply. Yup you're right that these seem to be separate issues 👍

**Comment by christopherfujino** [comment]
> Ah I missed that log line in your reply. Yup you're right that these seem to be separate issues 👍

Please file a new bug though, and I'd be happy to discuss :)

**Comment by loic-sharma** [comment]
Yup, just opened this here: https://github.com/flutter/flutter/issues/132592
